src/applications/managing-app-configuration.md

```diff
@@ -54,68 +54,106 @@ Detachments can be performed with `deis config:unset`.
 
 ## Custom Health Checks
 
-By default, Workflow only checks that your application containers start in
-their Pod. If you would like Kubernetes to respond to appliation health, you
-may add a health check by configuring URL, initial delay, and timeout
-values.
+By default, Workflow only checks that the application starts in their Pod. If it is preferred to
+have Kubernetes respond to application health, a health check may be added by configuring a health
+check probe for the application.
 
-The health checks are implemented as [Kubernetes container probes][kubernetes-probes]. Currently only HTTP GET is supported.
+The health checks are implemented as [Kubernetes container probes][kubernetes-probes]. A `liveness`
+and a `readiness` probe can be configured, and each probe can be of type `httpGet` or `exec`,
+depending on the type of probe the container requires.
 
-Available configuration options for health checks are the following, including defaults:
+A liveness probe is useful for applications running for long periods of time, eventually
+transitioning to broken states and cannot recover except by restarting them.
 
-**HEALTHCHECK_URL**
+Other times, a readiness probe is useful when the container is only temporarily unable to serve,
+and will recover on its own. In this case, if a container fails its readiness probe, the container
+will not be shut down, but rather the container will stop receiving incoming requests.
 
-Path in the application to use for health check, such as /healthz - Query Parameters are not supported.
+`httpGet` probes are just as it sounds: it performs a HTTP GET operation on the Pod. A response
+code inside the 200-399 range is considered a pass.
 
-This value needs to be set for any health check to happen. Needs to accept a HTTP GET request and return a HTTP status between 200-399.
+`exec` probes run a command inside the Pod to determine its health. An exit code of zero is
+considered a pass, while a non-zero status code is considered a fail.
 
-**HEALTHCHECK_TIMEOUT**
+Health checks can be configured on a per-application basis using `deis healthchecks:set`. To
+configure a `httpGet` liveness probe:
 
-Number of seconds after which the health check times out. Defaults to 50 seconds
-
-**HEALTHCHECK\_INITIAL_DELAY**
-
-Number of seconds before health checks starts checking the application after the start of the pod.
-
-This is useful to set if the application takes a while to get setup, due to migrations, cache warming or otherwise. Prevents Kubernetes (and Deis Workflow) from replacing an application pod in its setup phase.
-
-Defaults to 50 seconds
+```
+$ deis healthchecks:set liveness httpGet / 80 --initial-delay-timeout 30 --timeout-seconds 1
```

Comment:
like `deis healthchecks:set liveness httpGet 80 --path /index.html` how we are treating other optional parameters